service/tablet_allocator.cc

```diff
@@ -25,108 +25,6 @@ namespace service {
 
 seastar::logger lblogger("load_balancer");
 
-struct load_balancer_dc_stats {
```

Comment:
Code movement (without logic changes) should be in one commit for easy review.

Reply:
See b3369611bceac38eb5caddde79697522bef49fb4 for an example.

Reply:
We'll split this PR into 2:
1. here we'll only leave the request processing path and stub `reallocate_tablets_for_new_rf`
2. in https://github.com/scylladb/scylladb/pull/17116 @KrzaQ will implement `reallocate_tablets_for_new_rf`, 

I'll remove the 2nd part from this PR and @KrzaQ will address this comment in https://github.com/scylladb/scylladb/pull/17116